lib/bridge.js

```diff
@@ -636,7 +636,7 @@ Bridge.prototype.provisionUser = function(matrixUser, provisionedUser) {
             return newUser.setAvatarUrl(provisionedUser.url);
         });
     }
-    if (!provisionedUser.dontJoinGroup && self.groupId) {
+    if (provisionedUser.shouldJoinGroup !== false && self.groupId) {
```

Comment:
Maybe have a comment to explain that this is effectively default `true`?

Reply:
Actually, given that you've done it the explicit way for `shouldAdvertiseGroup`, I'd just do them both the same way for readability and consistency.